### PR TITLE
cli: refactor AddressFlag usages

### LINKED ISSUE #1895: Wallet signing command doesn't accept hashes for addresses
```
$ ./bin/neo-go wallet sign --in h.json -r http://localhost:20332 -w oracle-con.wallet -a 0xfc0b5a05ddde5773671dbabbb0c62461074a3a9e
invalid address: invalid base58 digit ('0')
$ ./bin/neo-go wallet sign --in h.json -r http://localhost:20332 -w oracle-con.wallet -a NaLbn4u9rZzk2gifu9hnPtyUbE8mtYZVmT
Password > 
2a49ad1165547792d774a19f1759237a27f21f3110612ec47e598222b631cb3e
```
Addresses and script hashes are essentially the same data, everywhere we accept one, the other should be accepted too.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
### Problem

We have a lot of places where `cli.StringFlag` is used instead of `flags.AdddressFlag`.

Also close #1895.

### CODE REVIEW

File: cli/wallet/wallet.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func NewCommands() []cli.Command {
 				Flags: append([]cli.Flag{
 					walletPathFlag,
 					wifFlag,
-					cli.StringFlag{
+					flags.AddressFlag{
 						Name:  "contract, c",
 						Usage: "Contract hash",
```
Comment (roman-khimov) [inline cli/wallet/wallet.go:202]: or address

File: cli/nep17_test.go [hunk 2]
```diff
@@ -200,14 +200,21 @@ func TestNEP17ImportToken(t *testing.T) {
...
+	// missing token hash
+	e.RunWithError(t, "neo-go", "wallet", "nep17", "import",
+		"--rpc-endpoint", "http://"+e.RPC.Addr,
+		"--wallet", walletPath)
+
 	e.Run(t, "neo-go", "wallet", "nep17", "import",
 		"--rpc-endpoint", "http://"+e.RPC.Addr,
 		"--wallet", walletPath,
 		"--token", gasContractHash.StringLE())
+
```
Comment (roman-khimov) [inline cli/nep17_test.go:213]: Whitespace change.

**Review by roman-khimov — COMMENTED** [review]
Do we have any `flags.ParseAddress` use cases left after this PR?

**Comment by AnnaShaleva** [comment]
> Do we have any `flags.ParseAddress` use cases left after this PR?

Yes, we use it to parse cosigners and unnamed contract scripthash parameter from `invoke*` CLI calls.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**